compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/inner-function/nullable-objects/array-map-simple.expect.md

```diff
@@ -0,0 +1,111 @@
+
+## Input
+
+```javascript
+/**
+ * Test that we're not hoisting property reads from lambdas that are created to
+ * pass to opaque functions, which often have maybe-invoke semantics.
+ *
+ * In this example, we shouldn't hoist `arr[0].value` out of the lambda.
```

Comment:
Since arr1 and arr2 are props, shouldn't they be deeply immutable? It should be safe to hoist them out?

If this is ever unsafe it's because of a rules-of-react violation

Reply:
Yep arr1 and arr2 are both known to be deeply immutable, but that doesn't mean it's safe to hoist property loads from them as they might be null (or have no elements)